avocado/core/xunit.py

```diff
@@ -87,9 +87,9 @@ def add_success(self, state):
         :type state: dict
         """
         tc = '\t<testcase classname={class} name={name} time="{time}"/>'
-        values = {'class': self._escape_attr(state['class_name']),
-                  'name': self._escape_attr(state['tagged_name']),
-                  'time': state['time_elapsed']}
+        values = {'class': self._escape_attr(state.get('class_name', "<unknown>")),
+                  'name': self._escape_attr(state.get('tagged_name', "<unknown>")),
+                  'time': state.get('time_elapsed', -1)}
```

Comment:
Here `-1` is being used for `time_elapsed`, which I think is more appropriate as suggested before.